--- linit.c	Thu Dec 27 13:02:25 2007
+++ linit_.c	Sun Aug 31 16:14:14 2008
@@ -13,6 +13,7 @@
 #include "lualib.h"
 #include "lauxlib.h"
 
+int luaopen_istools( lua_State *L );
 
 static const luaL_Reg lualibs[] = {
   {"", luaopen_base},
@@ -23,6 +24,7 @@
   {LUA_STRLIBNAME, luaopen_string},
   {LUA_MATHLIBNAME, luaopen_math},
   {LUA_DBLIBNAME, luaopen_debug},
+  {"istools", luaopen_istools},
   {NULL, NULL}
 };
 
